content/docs/command-reference/add.md

```diff
@@ -141,12 +140,17 @@ directory symlinks.
   the given target. This option allows to set the name and the path of the
   generated `.dvc` file.
 
+- `--glob` - allows adding files and directories that match the
+  [pattern](https://docs.python.org/3/library/glob.html) specified in `targets`.
+  Shell style wildcards supported: `*`, `?`, `[seq]`, `[!seq]`, and `**`
+
 - `--external` - allow `targets` that are outside of the DVC repository. See
   [Managing External Data](/doc/user-guide/managing-external-data).
 
-- `--glob` - allows adding files and directories that match the specified
-  pattern as specified by `target`. Shell-style wildcards are supported: `*`,
-  `?`, `[seq]`, `[!seq]`, and `**`.
+  > Note that external outputs require an external cache setup (unless
+  > `--no-commit` is used), and are never pushed or pulled from/to
```

Comment:
is there a reasonable workflow with `--no-commit` + `--external`?

Reply:
I think so, especially since we mention "For cached external outputs..." in https://dvc.org/doc/user-guide/managing-external-data

For example if you want to avoid the external data from being moved and linked back to its position for other processes that also use it which may not support handling links.

Reply:
okay, we don't move data, but we also don't calculate any hashes in this case, right? what would the next step for this `.dvc` file?

Reply:
I just tried this and it does calculate the hash. The data is just not moved to the cache. (At least using an external data file in the local system)

Reply:
And fortunately the --no-commit option description doesn't state that the has isn't calculated.

Unrelated: It does however mention the state file (fixed the path in 2ab35cc2) which is probably completely out of context/unnecessary to mention here but oh well...

Reply:
`I just tried this and it does calculate the hash` - hmm looks like a bug to me. Local `--no-commit` avoids calculating hash for outputs as far as I remember. Let me check.

Reply:
It's calculating it even for regular outputs in the workspace for me (DVC 1.9.1 on Win).

Reply:
okya, you are right on this (my usual confusion since --no-commit is not an opposite to `dvc commit` unfortunately)

it calculates hashes

```
√ Projects/test-dvc % dvc add --external --no-commit s3://dvc-temp/ivan/images
Adding...
ERROR: Current operation was unsuccessful because 's3://dvc-temp/ivan/images' requires existing cache on 's3' remote. See <https://man.dvc.org/config#cache> for information on how to set up remote cache.
```

it is still not correct it seems.

To be honest I doubt that this note could be understood by anyone :) I would keep this information to the Managing external data section.

Reply:
> It does however mention the state file

it is irrelevant, agreed

Reply:
> I would keep this information to the Managing external data section
> it (state file) is irrelevant, agreed

Ok, updated both things in 7bad093e. Just left the note about the external cache, which still seems pretty relevant IMO.

Reply:
PTAL